### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert doesn't consider nested classes

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3614]
**Affects PMD Version:** 6.40.0

**Rule:** [JUnitTestsShouldIncludeAssert](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert)

**Description:** JUnit5 supports defining [nested test classes](https://junit.org/junit5/docs/current/user-guide/#writing-tests-nested). These should also be considered by the rule.

**Code Sample demonstrating the issue:**

```java
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

public class BigTest {
    @Nested
    class InnerTestCases {
        @Test
        void check() {
            // empty, missing assert
        }
    }
}
```

**Expected outcome:**

PMD should report a violation at line 8, but doesn't. This is a false-negative.

Note - found via #2687
